### ISSUE TITLE
[java] JUnitTestsShouldIncludeAssert false positive

### ISSUE THREAD

**Issue by cesares-basilico** [pmd/pmd#631]
**Rule Set:** pmd-unit-tests:JUnitTestsShouldIncludeAssert

**Description:** I got this false positive, the assertion in implicit with the expected exception rule.

**Code Sample demonstrating the issue:**

```
@Rule
public ExpectedException expectedEx = ExpectedException.none();

@Test
public void testCheckStateBoOperatorNoCsId() {
  expectedEx.expect(InvalidUserException.class);
  expectedEx.expect(ErrorMatcher.hasCode(ErrorCodes.XXX));
  xxx.yyy();
}
```

**Running PMD through:** *Maven*

**Comment by jsotuyod** [comment]
@cesares-basilico What version of PMD are you using? This looks like a duplicate of #285, which has been fixed since PMD 5.4.6.

Bare in mind the latest release is 5.8.1, with 6.0.0 coming out next.

**Comment by cesares-basilico** [comment]
I'm using last version of maven-pmd-plugin which uses PMD 5.6.1 as stated on their site

**Comment by jsotuyod** [comment]
@cesares-basilico I can't reproduce it under the current build. It's possible it was addressed in the many fixes and improvements to both the rule itself and the symbol table code since 5.6.1.

Would you mind [changing your pom to use PMD 5.8.1](https://pmd.github.io/pmd/pmd_userdocs_tools_maven.html) and try it yourself?

**Comment by cesares-basilico** [comment]
Version 5.8.1 seems working fine, thanks for your help and suggestions

**Comment by jsotuyod** [comment]
@cesares-basilico great! I'm closing this then!
